Core/CoreTests/Login/LoginStartViewControllerTests.swift

```diff
@@ -210,6 +210,28 @@ class LoginStartViewControllerTests: CoreTestCase {
         XCTAssertFalse(controller.useQRCodeButton.isHidden)
         XCTAssertTrue(controller.useQRCodeDivider.isHidden)
     }
+
+    func testDisabledButtonsWhenOffline() throws {
+        controller = LoginStartViewController.create(loginDelegate: self, fromLaunch: true, app: .student, offlineModeInteractor: OfflineModeInteractorMock(mockIsInOfflineMode: true))
+        controller.view.layoutIfNeeded()
+
+        controller.canvasNetworkButton.sendActions(for: .primaryActionTriggered)
+        XCTAssert(router.viewControllerCalls.last?.0 is UIAlertController)
+        var alert = try XCTUnwrap(router.presented as? UIAlertController)
+        XCTAssertEqual(alert.title, "Offline mode")
+
+        controller.findSchoolButton.sendActions(for: .primaryActionTriggered)
+        alert = try XCTUnwrap(router.presented as? UIAlertController)
+        XCTAssertEqual(alert.title, "Offline mode")
+
+        controller.lastLoginButton.sendActions(for: .primaryActionTriggered)
+        alert = try XCTUnwrap(router.presented as? UIAlertController)
+        XCTAssertEqual(alert.title, "Offline mode")
+
+        controller.useQRCodeButton.sendActions(for: .primaryActionTriggered)
+        alert = try XCTUnwrap(router.presented as? UIAlertController)
+        XCTAssertEqual(alert.title, "Offline mode")
```

Comment:
Shouldn't we reset the mock router's `presented` property? If `findSchoolButton`, `lastLoginButton` or `useQRCodeButton` don't present an alert the test will pass since the `canvasNetworkButton` already presented an alert.